### PR TITLE
test: enable spec tests related to eip-7549

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Increase spec test coverage for Electra.

**Description**

- Enable the spec tests related to EIP7549. 
- Add `SPEC_FILTER_FORK` and `SPEC_FILTER_FORK` environmental variable which will be read in during spec test
- Fix committeeIndex check during `processAttestation()`

**Steps to test or reproduce**

- Run all tests

### CODE REVIEW

File: packages/state-transition/src/block/processAttestationPhase0.ts [hunk 1]
```diff
@@ -93,7 +93,7 @@ export function validateAttestation(
       throw new Error(`AttestationData.index must be zero: index=${data.index}`);
     }
     const attestationElectra = attestation as electra.Attestation;
-    const committeeBitsLength = attestationElectra.committeeBits.bitLen;
+    const committeeBitsLength = attestationElectra.committeeBits.getTrueBitIndexes().length;
```
Comment (wemeetagain) [inline packages/state-transition/src/block/processAttestationPhase0.ts]: @g11tech is this correct?

Comment (nazarhussain) [reply packages/state-transition/src/block/processAttestationPhase0.ts]: The `committeeBits` is a bit vector so the `bitLen` will always remain fixed which is `MAX_COMMITTEES_PER_SLOT`. Hence if a chain is starting with initial validators `less than MAX_COMMITTEES_PER_SLOT` the attestation check will never pass. 

So we actually need to get how many comittees actually participatd using `getTrueBitIndexes`.

File: packages/beacon-node/test/spec/utils/specTestIterator.ts [hunk 2]
```diff
@@ -57,14 +58,17 @@ const coveredTestRunners = [
...
   // TODO: capella
   // BeaconBlockBody proof in lightclient is the new addition in v1.3.0-rc.2-hotfix
   // Skip them for now to enable subsequently
-  skippedPrefixes: [
-    "capella/light_client/single_merkle_proof/BeaconBlockBody",
-    "deneb/light_client/single_merkle_proof/BeaconBlockBody",
+  skippedTestSuites: [
+    /^capella\/light_client\/single_merkle_proof\/BeaconBlockBody.*/,
+    /^deneb\/light_client\/single_merkle_proof\/BeaconBlockBody.*/,
+    // /^electra\/(?!operations\/attestations)(?!operations\/attester_slashing)/,
```
Comment (ensi321) [inline packages/beacon-node/test/spec/utils/specTestIterator.ts:68]: What's the purpose of having this line? Do we want to skip attester slashing or not?

File: packages/state-transition/src/block/processAttestationPhase0.ts (hunk 1 above)
Comment (ensi321) [reply packages/state-transition/src/block/processAttestationPhase0.ts]: The `if(committeeBitsLength > committeeCount)` check in electra-fork branch is wrong. 

The check is supposedly to be identical to this [line](https://github.com/ethereum/consensus-specs/blob/812ac2ce8fd436c757bb22d4aa5c90bf4923b664/specs/electra/beacon-chain.md?plain=1#L1067) of the spec. The reason is although committeeBits has a set length of 64, actual number of committees might be less than that. If the 60th bit is set, but there is only 40 committees, then this check needs to throw error

Here we actually want the position of the last set bit in `attestationElectra.committeeBits`. If `lastCommitteeIndex` is greater than `committeeCount`, then we throw error.

Comment (ensi321) [reply packages/state-transition/src/block/processAttestationPhase0.ts]: So something like this:
```
    const committeeIndices = attestationElectra.committeeBits.getTrueBitIndexes();
    if (committeeIndices.length === 0) {
        throw Error("Should have at least one committee bit set");
    } else {
        const lastCommitteeIndex = committeeIndices[committeeIndices.length - 1];
        if (lastCommitteeIndex >= committeeCount) {
            throw new Error("Committee index exceeds committee count");
         }
    }
```

This is beyond the scope of this PR, happy to fix this in another PR. Maybe add a todo here for now. Thanks for catching it @nazarhussain @wemeetagain

File: packages/beacon-node/test/spec/utils/specTestIterator.ts (hunk 2 above)
Comment (nazarhussain) [reply packages/beacon-node/test/spec/utils/specTestIterator.ts:68]: Just a reference to remember, how to enable multiple operaitons. As I was working on those.

**Comment by nazarhussain** [comment]
@ensi321 As this PR is enabling the spec tests for `electra/oeprations/attestations` so any fix to make these pass preferrably go with this PR. So I just added a commit with the fix you suggested and it's passing the spec.

File: packages/beacon-node/test/spec/utils/specTestIterator.ts [hunk 3]
```diff
@@ -100,7 +104,10 @@ export function specTestIterator(
   opts: SkipOpts = defaultSkipOpts
 ): void {
   for (const forkStr of readdirSyncSpec(configDirpath)) {
-    if (opts?.skippedForks?.includes(forkStr)) {
+    if (
+      opts?.skippedForks?.includes(forkStr) ||
+      (process.env.SPEC_FILTER_FORK && forkStr !== process.env.SPEC_FILTER_FORK)
```
Comment (ensi321) [inline packages/beacon-node/test/spec/utils/specTestIterator.ts:109]: Curious why we would want to specify fork that we want to skip in a env variable `SPEC_FILTER_FORK`. What's the use case for this? I think `defaultSkipOpts` is serving quite well

**Review by ensi321 — COMMENTED** [review]
looks good overall minor comment

File: packages/beacon-node/test/spec/utils/specTestIterator.ts (hunk 3 above)
Comment (nazarhussain) [reply packages/beacon-node/test/spec/utils/specTestIterator.ts:109]: Quickly run the spec tests for some forks. Whilde debugging, if we run for all forks it take too much time. 

```bash
SPEC_FILTER_FORK=electra yarn test:spec:minimal
```

**Review by ensi321 — APPROVED** [review]
lgtm

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.22.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.22.0) :tada:

**Pull request merged**